Source/WebCore/platform/graphics/gstreamer/eme/WebKitOpenCDMSessionEncKey.cpp

```diff
@@ -0,0 +1,110 @@
+
+/*
+ * Copyright (C) 2016-2017 TATA ELXSI
+ * Copyright (C) 2016-2017 Metrological
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ * "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ * LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A
+ * PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+ * HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+ * SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+ * LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#include "config.h"
+#include "WebKitOpenCDMSessionEncKey.h"
+
+#if ENABLE(LEGACY_ENCRYPTED_MEDIA) && USE(OCDM)
+
+#include "CDM.h"
+#include "CDMSession.h"
+#include <gst/gst.h>
+#include "WebKitMediaKeyError.h"
+
+GST_DEBUG_CATEGORY_EXTERN(webkit_media_opencdm_decrypt_debug_category);
+#define GST_CAT_DEFAULT webkit_media_opencdm_decrypt_debug_category
+
+namespace WebCore {
+
+WebKitOpenCDMSessionEncKey::WebKitOpenCDMSessionEncKey(CDMSessionClient* client, OpenCdm* openCdm)
+    : m_client(client)
+    , m_openCdmSession(openCdm)
+    , m_destinationUrlLength(0)
+{
+}
+
+RefPtr<Uint8Array> WebKitOpenCDMSessionEncKey::generateKeyRequest(const String& mimeType, Uint8Array* initData, String& destinationUrl, unsigned short& errorCode, uint32_t&)
+{
+    int returnValue = 0;
+
+    unsigned char initDataValue[initData->length()];
+
+    memcpy(initDataValue, initData->data(), initData->length());
+
+    string sessionId;
+    m_openCdmSession->CreateSession(mimeType.utf8().data(), initDataValue,
+        initData->length(), sessionId);
+    if (!sessionId.size()) {
+        GST_WARNING("SessionId is empty\n");
+        return nullptr;
+    }
+    m_sessionId = String::fromUTF8(sessionId.c_str());
+
+    unsigned char temporaryUrl[1024] = "\0";
```

Comment:
This shouldn't type properly because you're a const char[2] to an unsigned char[1024]. If you want to ensure that temporaryUrl is initialized as null-terminated I think the proper way of doing this is:

unsigned char temporaryUrl1024];
temporaryUrl[0] = '\0';